crates/bitcoin-da/src/service.rs

```diff
@@ -546,9 +546,9 @@ impl DaService for BitcoinService {
     async fn get_block_by_hash(&self, hash: [u8; 32]) -> Result<Self::FilteredBlock, Self::Error> {
```

Comment:
It would be better to also accept BlockHash as an arg

Reply:
needs change on BlockHashTrait. too much work for now.